app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableQuantityCondition.php

```diff
@@ -7,35 +7,23 @@
 
 namespace Magento\InventorySales\Model\IsProductSalableForRequestedQtyCondition;
 
-use Magento\InventoryReservationsApi\Model\GetReservationsQuantityInterface;
 use Magento\InventorySalesApi\Api\IsProductSalableForRequestedQtyInterface;
 use Magento\InventorySalesApi\Model\GetStockItemDataInterface;
 use Magento\InventorySalesApi\Api\Data\ProductSalableResultInterface;
 use Magento\InventorySalesApi\Api\Data\ProductSalableResultInterfaceFactory;
 use Magento\InventorySalesApi\Api\Data\ProductSalabilityErrorInterfaceFactory;
-use Magento\InventoryConfigurationApi\Api\GetStockItemConfigurationInterface;
-use Magento\InventoryConfigurationApi\Api\Data\StockItemConfigurationInterface;
+use Magento\InventorySalesApi\Api\GetProductSalableQtyInterface;
 
 /**
  * @inheritdoc
  */
-class IsSalableWithReservationsCondition implements IsProductSalableForRequestedQtyInterface
+class IsSalableQuantityCondition implements IsProductSalableForRequestedQtyInterface
```

Comment:
This seems to be a massive change, how would we calculate reservations now?

Reply:
The same discussion as above really. An attempt to standardise on the use of the SalableQuantity function to determine the quantity of stock for sale, instead of calculating it directly using stock levels and reservations (thats the job of the salableQuantity function)

If you work through the logic of the changes below and compare with whats going on inside getSalableQty its the same thing.

You can still access reservations directly, but if all that you want to know is how many can I sell (i.e the salable quantity) then use the salableQuantity functions (which takes into account current reservations)

Again, open for discussion. It just feels cleaner, more correct, less duplication of logic by making everything use getSalableQty instead of working directly with stock levels and reservations.

Reply:
Happy to take the chat over to slack #msi if requred